### PR TITLE
[model] refactor: add model patch for deepseek-v3

### PR DESCRIPTION
**Pull request by FoolPlayer**
### What does this PR do?

add model patch for deepseek-v3

### Checklist Before Starting

- [ ] Search for similar PRs. Paste at least one query link here: ...
- [ ] Format the PR title as `[{modules}] {type}: {description}` (This will be checked by the CI)
  - `{modules}` include `misc`, `ci`, `config`, `docs`, `data`, `dist`, `omni`, `logging`, `model`, `optim`, `ckpt`, `release`, `task`, `perf`, `ops`, `parallel`
  - If this PR involves multiple modules, separate them with `,` like `[ci, data, model]`
  - `{type}` is in `feat`, `fix`, `refactor`, `chore`, `test`
  - If this PR breaks any API (CLI arguments, config, function signature, etc.), add `[BREAKING]` to the beginning of the title.
  - Example: `[BREAKING][parallel, model] feat: dynamic batching`

### Test

> For changes that can not be tested by CI (e.g., algorithm implementation, new model support), validate by experiment(s) and show results like training curve plots, evaluation results, etc.

### API and Usage Example

> Demonstrate how the API changes if any, and provide usage example(s) if possible.

```python
# Add code snippet or script demonstrating how to use this
```

### Design & Code Changes

> Demonstrate the high-level design if this PR is complex, and list the specific changes.

### Checklist Before Submitting

> [!IMPORTANT]
> Please check all the following items before requesting a review, otherwise the reviewer might deprioritize this PR for review.

- [ ] Read the [Contribute Guide](https://github.com/ByteDance-Seed/VeOmni/blob/main/CONTRIBUTING.md).
- [ ] Apply [pre-commit checks](https://github.com/ByteDance-Seed/VeOmni/blob/main/CONTRIBUTING.md?plain=1#L22): `pre-commit install && pre-commit run --all-files --show-diff-on-failure --color=always`
- [ ] Add / Update [the documentation](https://github.com/ByteDance-Seed/VeOmni/blob/main/docs).
- [ ] Add unit or end-to-end test(s) to [the CI workflow](https://github.com/ByteDance-Seed/VeOmni/tree/main/.github/workflows) to cover all the code. If not feasible, explain why: ...

### CODE REVIEW

File: veomni/models/transformers/deepseek_v3/modeling_deepseek_v3.py [hunk 1]
```diff
@@ -0,0 +1,442 @@
...
+        orig_shape = hidden_states.shape
+        router_logits = self.gate(hidden_states)
+        topk_indices, topk_weights = self.route_tokens_to_experts(router_logits)
+        hidden_states = hidden_states.view(-1, hidden_states.shape[-1])
+        hidden_states = self.experts(hidden_states, topk_indices, topk_weights).view(*orig_shape)
+        hidden_states = hidden_states + self.shared_experts(residuals)
+        return hidden_states
+
+
+logger = logging.get_logger(__name__)
```
Comment (Coach257) [inline veomni/models/transformers/deepseek_v3/modeling_deepseek_v3.py]: mv to top

File: veomni/models/transformers/deepseek_v3/modeling_deepseek_v3.py [hunk 2]
```diff
@@ -0,0 +1,442 @@
...
+
+    if past_key_values is not None:
+        # sin and cos are specific to RoPE models; cache_position needed for the static cache
+        cache_kwargs = {"sin": sin, "cos": cos, "cache_position": cache_position}
+        key_states, value_states = past_key_values.update(key_states, value_states, self.layer_idx, cache_kwargs)
+
+    # --- Patch.1 ---
+    # Flash Attention requires Q/K and V to have the same head dimension on non-Hopper GPUs.
+    # For DeepSeek V3 MLA architecture where qk_head_dim != v_head_dim, we pad V to match Q/K.
+    _FA_IMPLS_NEED_PADDING = [
```
Comment (Coach257) [inline veomni/models/transformers/deepseek_v3/modeling_deepseek_v3.py]: from ..models.transformers.attention_utils import VARLEN_ATTENTION_TYPES

File: veomni/models/auto.py [hunk 3]
```diff
@@ -53,7 +53,7 @@ def build_config(config_path: str, **config_kwargs) -> "PretrainedConfig":
     """
     Builds the model config.
     """
-    return get_model_config(config_path, trust_remote_code=True, **config_kwargs)
+    return get_model_config(config_path, **config_kwargs)
```
Comment (Coach257) [inline veomni/models/auto.py]: why delete trust_remote_code

File: tests/models/test_models_patch.py
Comment (Luosuu) [inline tests/models/test_models_patch.py:1]: could you please help me to add dpsk v3 test cases in `test_trainer_saveload.py` to verify its EP+FSDP2 and DCP checkpointing. Thanks!

Comment (FoolPlayer) [reply tests/models/test_models_patch.py:1]: Sure, I'll add the test for deepseek_v3 in the next PR.
